src/Config/FileConfigManager.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace Spatie\Ignition\Config;
+
+use Spatie\Ignition\Contracts\ConfigManager;
+
+class FileConfigManager implements ConfigManager
+{
+    private const SETTINGS_FILE_NAME = '.ignition.json';
+
+    private string $path;
+
+    public function __construct(string $path = '')
+    {
+        $this->path = $this->initPath($path);
+    }
+
+    private function initPath(string $path): string
+    {
+        if ($path !== '') {
+            return $this->initPathFromArgument($path);
+        }
+
+        return $this->initPathFromEnvironment();
+    }
+
+    private function initPathFromArgument(string $path): string
+    {
+        if ($this->isValidPath($path)) {
+            return $path;
+        }
+
+        return '';
+    }
+
+    private function isValidPath(string $path): bool
+    {
+        if ($path === '') {
+            return false;
+        }
+
+        $path = realpath($path) ?: $path;
+
+        return file_exists($path) && is_writable($path);
+    }
+
+    protected function initPathFromEnvironment(): string
+    {
+        if ($this->isWindows()) {
+            if (empty($_SERVER['HOMEDRIVE']) || empty($_SERVER['HOMEPATH'])) {
+                return '';
+            }
+
+            $homeDirectory = $_SERVER['HOMEDRIVE'] . $_SERVER['HOMEPATH'];
+
+            return $this->preparePath($homeDirectory);
+        }
+
+        if ($homeDirectory = getenv('HOME')) {
+            return $this->preparePath($homeDirectory);
+        }
+
+        return '';
+    }
+
+    private function isWindows(): bool
+    {
+        return str_starts_with(strtoupper(PHP_OS), 'WIN');
+    }
+
+    public function load(): array
+    {
+        $file = $this->generateFullFilePath();
+
+        if (! $this->isValidFile($file)) {
+            return [];
+        }
+
+        $content = (string)file_get_contents($file);
+        $options = json_decode($content, true) ?? [];
+
+        return $options;
+    }
+
+    public function save(array $options): bool
+    {
+        $file = $this->generateFullFilePath();
+
+        if (! $this->isValidFile($file)) {
+            return false;
+        }
+
+        try {
+            file_put_contents($file, json_encode($options));
+        } catch (\Throwable) {
+            return false;
+        }
+
+        return true;
+    }
+
+    protected function isValidFile(string $file): bool
+    {
+        return file_exists($file) && @is_writable($file);
+    }
+
+    public function getSource(): array
+    {
+        return [
+            'name' => self::SETTINGS_FILE_NAME,
+            'path' => $this->path,
+            'file' => $this->generateFullFilePath(),
+        ];
+    }
+
+    protected function preparePath(string $path): string
+    {
+        return rtrim($path, DIRECTORY_SEPARATOR);
+    }
+
+    public function createSource(): bool
```

Comment:
Also, what do you think about calling this method from the `save()` method? We can still keep it public for use inside of tests tho.

Reply:
The idea to introduce and make `createSource()` method public was to give the possibility to create a settings storage somewhere outside. For example, if I don't want to use the `createSource` every time on the saving operation, I can just use it in a ServiceProvider to create it once. So, I'm still thinking about how to implement it and you are absolutely right that it's better to remove this call from `IgnitionConfig.php`.

I need to think about a better solution on how to create storage once and not use it on every save() call. Until then, I'll move it to the save() method.